src/utils/dispatchModal.ts

```diff
@@ -12,6 +13,24 @@ export enum ModalEvent {
     FilterDeleteModal = 'FilterDeleteModal',
 }
 
-export const dispatchModalEvent = (e: ModalEvent) => () => {
-    window.dispatchEvent(new Event(e));
+export interface MapModalToComponentProps {
+    [ModalEvent.GoalCreateModal]: GoalCreateFormProps;
```

Comment:
`ComponentProps<typeof GoalCreateForm>`

Reply:
if i add this import i will get the cycling dependency, because `dispatchModalEvent` uses in `GoalCreateForm` component

Reply:
Yep ;) Because event data and component props are different logical things